### PR TITLE
weylus: build from source and fix desktop shortcut

### PR DESCRIPTION
**Pull request by linsui**
###### Motivation for this change

I'm using weylus. I want to build it from source. The desktop icon is missing. I fix it.

###### Things done

1. build weylus from source
2. fix desktop icon
3. I thought it works for Mac since it's built from souce. But I don't have a Mac.
4. I haven't test it with pipewire. @noneucat Could you please help test it?

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

@legendofmiracles

### CODE REVIEW

**Comment by linsui** [comment]
@legendofmiracles Ping?

File: pkgs/applications/graphics/weylus/default.nix [hunk 1]
```diff
@@ -1,71 +1,77 @@
...
+    xorg.libXi
+    xorg.libXv
+    pango
+    libdrm
+  ];
+
+  nativeBuildInputs = [
+    cmake
+    nodePackages.typescript
+  ] ++ lib.optionals stdenv.isLinux [
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/weylus/default.nix:56]: Why isn't this made available on darwin?

Comment (linsui) [reply pkgs/applications/graphics/weylus/default.nix:56]: I don't have a Mac to test it. In [Readme](https://github.com/H-M-H/Weylus#building) it says that

> On Linux some additional dependencies are required to build Weylus. On Debian or Ubuntu they can be installed via:

So I guess these deps are not needed on Mac. But I don't really know.

**Comment by SuperSandro2000** [comment]
@ofborg eval
@ofborg build weylus

File: pkgs/applications/graphics/weylus/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/applications/graphics/weylus/default.nix:56]: Lets find out when ofborg is done.

File: pkgs/applications/graphics/weylus/default.nix [hunk 2]
```diff
@@ -1,71 +1,77 @@
...
+  postInstall = ''
+    install -vDm755 weylus.desktop $out/share/applications/weylus.desktop
   '';
 
   meta = with lib; {
     description = "Use your tablet as graphic tablet/touch screen on your computer";
     homepage = "https://github.com/H-M-H/Weylus";
     license = with licenses; [ agpl3Only ];
     maintainers = with maintainers; [ lom ];
-    platforms = [ "x86_64-linux" ];
```
[the remark this answers is not in the document]
Comment (linsui) [reply pkgs/applications/graphics/weylus/default.nix:69]: Ah, libva is only used on linux. https://github.com/H-M-H/Weylus/commit/b47e5f2818fac136fcdd839f01e2eeb3c46aceec

File: pkgs/applications/graphics/weylus/default.nix [hunk 3]
```diff
@@ -1,71 +1,77 @@
...
   buildInputs = [
-    libpng
-    dbus
-    libdrm
-    fontconfig
+    ffmpeg
+    x264
     libva
+  ] ++ lib.optionals stdenv.isLinux [
+    dbus
```
Comment (linsui) [inline pkgs/applications/graphics/weylus/default.nix:36]:
```suggestion
  ] ++ lib.optionals stdenv.isLinux [
    dbus
    libva
```

File: pkgs/applications/graphics/weylus/default.nix [hunk 4]
```diff
@@ -1,71 +1,77 @@
...
-    copyDesktopItems ./weylus.desktop
-
-    runHook postInstall
-  '';
-
   buildInputs = [
-    libpng
+    ffmpeg
+    x264
+  ] ++ lib.optionals stdenv.isLinux [
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/weylus/default.nix:34]:
```suggestion
  ] ++ lib.optionals stdenv.isDarwin [
    ApplicationServices
    Carbon
    Cocoa
  ] ++ lib.optionals stdenv.isLinux [
```

and something like 

```nix
graphviz = callPackage ../tools/graphics/graphviz {
  inherit (darwin.apple_sdk.frameworks) ApplicationServices Carbon Cocoa;
};
```

in top-level.nix

File: pkgs/applications/graphics/weylus/default.nix [hunk 5]
```diff
@@ -1,71 +1,77 @@
...
-, libpng
-, pipewire
-, makeWrapper
-, autoPatchelfHook
+, pkg-config
+, pango
+, cmake
+, autoconf
+, libtool
+, nodePackages
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/weylus/default.nix:17]:
```suggestion
, nodePackages
, ApplicationServices
, Carbon
, Cocoa
```

**Comment by legendofmiracles** [comment]
Result of `nixpkgs-review pr 161052` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>weylus</li>
  </ul>
</details>

File: pkgs/applications/graphics/weylus/default.nix [hunk 6]
```diff
@@ -1,71 +1,84 @@
...
+, ApplicationServices
+, Carbon
+, Cocoa
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/weylus/default.nix:20]:
```suggestion
, Cocoa
, VideoToolbox
```

File: pkgs/top-level/all-packages.nix [hunk 7]
```diff
@@ -1032,7 +1032,9 @@ with pkgs;
 
   weidu = callPackage ../tools/games/weidu { };
 
-  weylus = callPackage ../applications/graphics/weylus { };
+  weylus = callPackage ../applications/graphics/weylus  {
+    inherit (darwin.apple_sdk.frameworks) ApplicationServices Carbon Cocoa;
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:1036]:
```suggestion
    inherit (darwin.apple_sdk.frameworks) ApplicationServices Carbon Cocoa VideoToolbox;
```

**Comment by SuperSandro2000** [comment]
Yes, it builds. I don't know if it fully works but at least better than nothing.

File: pkgs/applications/graphics/weylus/default.nix [hunk 8]
```diff
@@ -1,71 +1,86 @@
...
+    autoconf
+    libtool
+  ];
+
+  cargoSha256 = "1pigmch0sy9ipsafd83b8q54xwqjxdaif363n1q8n46arq4v81j0";
+
+  cargoBuildFlags = [ "--features=ffmpeg-system" ];
+  cargoTestFlags = [ "--features=ffmpeg-system" ];
 
-  postFixup = let
```
Comment (zhaofengli) [inline pkgs/applications/graphics/weylus/default.nix:55]: GST_PLUGIN_PATH is still required (https://github.com/H-M-H/Weylus/issues/155#issuecomment-1120149472). I've opened #174434 to re-add it on Linux.

**Pull request merged**